### ISSUE TITLE
Add more descriptive error message for unwrap call

### ISSUE THREAD

**Issue by frewsxcv** [bevyengine/bevy#3769]
I am trying to upgrade my application to bevy 0.6 and I'm hitting a panic on this `unwrap` call:

https://github.com/bevyengine/bevy/blob/ac63c491fb0d6bd8852e0d7e57280d4c3da35faa/crates/bevy_render/src/render_phase/draw.rs#L318

Does this mean that I forgot to add a component to my system? If so, how do you feel about replacing that `unwrap` with an `expects("......")` that is more actionable for the user?

**Comment by alice-i-cecile** [comment]
Yep, this would be much better served by an expect. The failure here is that the resource isn't added. Let me investigate why that might have happened quick...

**Comment by alice-i-cecile** [comment]
There's a few of these around: searching for `.init_resource::<DrawFunctions` shows me three in `bevy_core_pipeline`, one in `bevy_pbr`, and another in `bevy_ui`. In your particular case, I suspect that you're missing one of the corresponding plugins.

I'm not actually entirely sure what improvements we could make over the unwrap error message. Perhaps reporting the concrete type of `P` would be helpful in this particular case?
